cli/hzn.go

```diff
@@ -352,17 +352,31 @@ Environment Variables:
-	policyCompNodeArch := policyCompCmd.Flag("arch", msgPrinter.Sprintf("The architecture of the node. It is required when --node-pol is specified. If omitted, the service of all the architectures referenced in the business policy will be checked for compatibility.")).Short('a').String()
 	policyCompBPolId := policyCompCmd.Flag("business-pol-id", msgPrinter.Sprintf("The Horizon exchange business policy ID. Mutually exclusive with --business-pol. If you don't prepend it with the organization id, it will automatically be prepended with the node's organization id.")).Short('b').String()
+	policyCompNodeArch := policyCompCmd.Flag("arch", msgPrinter.Sprintf("The architecture of the node. It is required when -n is not specified. If omitted, the service of all the architectures referenced in the business policy will be checked for compatibility.")).Short('a').String()
 	policyCompNodePolFile := policyCompCmd.Flag("node-pol", msgPrinter.Sprintf("The JSON input file name containing the node policy. Mutually exclusive with -n.")).String()
 	policyCompBPolFile := policyCompCmd.Flag("business-pol", msgPrinter.Sprintf("The JSON input file name containing the business policy. Mutually exclusive with -b.")).String()
 	policyCompSPolFile := policyCompCmd.Flag("service-pol", msgPrinter.Sprintf("(optional) The JSON input file name containing the service policy. If omitted, the service policy will be retrieved from the exchange for the service defined in the business policy.")).String()
-	policyCompCheckAll := policyCompCmd.Flag("check-all", msgPrinter.Sprintf("Show the compatibility status of all the service versions referenced in the business policy.")).Short('c').Bool()
-	policyCompLong := policyCompCmd.Flag("long", msgPrinter.Sprintf("Show policies used for the compatibility checking.")).Short('l').Bool()
-	//userinputCompCmd := deploycheckCmd.Command("userinput", msgPrinter.Sprintf("Check user input compatibility."))
-	//allCompCmd := deploycheckCmd.Command("all", msgPrinter.Sprintf("Check all compatibilities for a deployment."))
+	userinputCompCmd := deploycheckCmd.Command("userinput", msgPrinter.Sprintf("Check user input compatibility."))
+	userinputCompNodeId := userinputCompCmd.Flag("node-id", msgPrinter.Sprintf("The Horizon exchange node ID. Mutually exclusive with --node-ui. If omitted, the node ID that the current device is registered with will be used. If you don't prepend it with the organization id, it will automatically be prepended with the -o value.")).Short('n').String()
+	userinputCompBPolId := userinputCompCmd.Flag("business-pol-id", msgPrinter.Sprintf("The Horizon exchange business policy ID. Mutually exclusive with --business-pol. If you don't prepend it with the organization id, it will automatically be prepended with the node's organization id.")).Short('b').String()
```

Comment:
Are you also going to support userinput comparisons for patterns?

Reply:
Good question. I think we should since this is a separate command/api from policy compatibility. I will add it later after I am done with the policy. I will put a work item in issue 1326 for this.